src/main/java/seedu/duke/data/TransactionList.java

```diff
@@ -265,6 +272,98 @@ public String listCategoricalSavings() {
         return categoricalSavingsList;
     }
 
+    /**
+     * Reads the transactions list and adds each amount to the month and year in monthly savings hashmap.
+     *
+     * @param monthlyExpenditure A hashmap containing all month-expenditure pair for total expenditure and savings.
+     * @return A hashmap containing all month-expenditure pair for total expenditure and savings.
+     */
+    public HashMap<String, int[]> processMonthlyExpenditure(HashMap<String, int[]> monthlyExpenditure)
+            throws InputTransactionInvalidTypeException {
+        for (Transaction transaction : transactions) {
+            String date = transaction.getDate().format(DateTimeFormatter.ofPattern(DATE_MONTH_PATTERN.toString()));
+            int income = 0;
+            int expense = 0;
+
+            // Checks whether transaction is Income or Expense and adds to respective amount
+            try {
+                if (isTransactionInstance(transaction, CLASS_TYPE_INCOME)) {
+                    income = transaction.getAmount();
+                } else {
+                    expense = - transaction.getAmount();
+                }
+            } catch (ClassNotFoundException e) {
+                throw new InputTransactionInvalidTypeException();
+            }
+
+            // Creates a new month and year with starter amounts if not exists in hashmap
+            if (!monthlyExpenditure.containsKey(date)) {
+                monthlyExpenditure.put(date, new int[]{income, expense, income + expense});
+                continue;
+            }
+
+            // Adds amounts to existing month and year in hashmap
+            int updatedIncome = monthlyExpenditure.get(date)[0] + income;
+            int updatedExpense = monthlyExpenditure.get(date)[1] + expense;
+
+            monthlyExpenditure.put(date, new int[]{updatedIncome, updatedExpense,
+                updatedIncome + updatedExpense});
+        }
+
+        return monthlyExpenditure;
+    }
+
+    /**
+     * Gets comment related to spending habit each month.
+     *
+     * @param savingsPercentage The percentage savings based against the total income.
+     * @return A string containing the comment related to the spending habit for the month.
+     */
+    public String getSpendingHabitComment(int savingsPercentage) {
+        System.out.println(savingsPercentage);
+        if (savingsPercentage >= 100) {
+            return INFO_STATS_MONTHS_COMMENT_ONE.toString();
+        } else if (savingsPercentage >= 75) {
+            return INFO_STATS_MONTHS_COMMENT_TWO.toString();
+        } else if (savingsPercentage >= 50) {
+            return INFO_STATS_MONTHS_COMMENT_THREE.toString();
+        } else if (savingsPercentage >= 25) {
```

Comment:
Magic numbers here for 100, 75, etc. I feel that they can be created as static variable such as 
`public static int HUNDRED_PERCENT = 100;`

Reply:
Thank you for the feedback. These constants have been renamed.